### PR TITLE
feat(roxctl): allow writing output and error streams to files

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The roxctl image is a bare image. It is the base image for some tekton actions.
Currently, tekton does not allow output redirection out of the box. In order to allow users to process roxctl output (e.g. open a ticket if CVEs are reported in an image scan output), it would be good to allow roxctl to redirect the standard output and the standard error without the standard shell output redirections.

The choice of environment variables rather than command line flags is driven by the fact that the roxctl environment (including output and error stream initialization) is setup before the command line flags are actually parsed.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/env/roxctl.go [hunk 1]
```diff
@@ -46,4 +46,10 @@ var (
 
 	// ScannerDBDownloadBaseURL specifies the base URL to use when downloading offline scanner definitions.
 	ScannerDBDownloadBaseURL = RegisterSetting("ROX_SCAN_DB_DL_BASE_URL", WithDefault("https://install.stackrox.io/scanner"))
+
+	// OutputFile specifies the path where roxctl should write its standard output
+	OutputFile = RegisterSetting("ROX_OUTPUT_FILE")
+
+	// ErrorFile specifies the path where roxctl should write its standard error
+	ErrorFile = RegisterSetting("ROX_ERROR_FILE")
```
Comment (0x656b694d) [inline pkg/env/roxctl.go:54]:
```suggestion
	OutputFile = RegisterSetting("ROX_STDOUT_FILE")

	// ErrorFile specifies the path where roxctl should write its standard error
	ErrorFile = RegisterSetting("ROX_STDERR_FILE")
```

**Review by 0x656b694d — COMMENTED** [review]
May we instead provide command-line parameters to the commands which output could be used in automation? This users could separate the processable generated output from logs, interactions and errors.
Like `roxctl image scan --output csv --output-file image_scan_results.csv`.

File: pkg/env/roxctl.go [hunk 2]
```diff
@@ -46,4 +46,10 @@ var (
 
 	// ScannerDBDownloadBaseURL specifies the base URL to use when downloading offline scanner definitions.
 	ScannerDBDownloadBaseURL = RegisterSetting("ROX_SCAN_DB_DL_BASE_URL", WithDefault("https://install.stackrox.io/scanner"))
+
+	// OutputFile specifies the path where roxctl should write its standard output
+	OutputFile = RegisterSetting("ROX_OUTPUT_FILE")
```
Comment (dhaus67) [inline pkg/env/roxctl.go:51]: nit: maybe also just call it `ROX_STDOUT/ROX_STDERR`, the file suffix can be dropped IMO.

File: roxctl/common/io/io.go [hunk 3]
```diff
@@ -34,13 +37,53 @@ func (i *ioImpl) ErrOut() io.Writer {
...
+			// Revert to default output
+			return nil
+		}
+		file, err := os.OpenFile(filePath, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0600)
+		if err != nil {
+			return nil
+		}
+		return file
+	}
+	if fileInfo.Mode()&0200 != 0 {
```
Comment (dhaus67) [inline roxctl/common/io/io.go:63]: What happens here if `err != nil && !os.IsNotExist(err)`? I assume `fileInfo` will be nil and `fileInfo.Mode()` will panic; I'd rather return in l.47 when `err != nil && !os.IsNotExist(err)`.

File: roxctl/common/io/io.go [hunk 4]
```diff
@@ -34,13 +37,53 @@ func (i *ioImpl) ErrOut() io.Writer {
 	return i.errOut
 }
 
+func getOutputFile(filePath string) *os.File {
+	// This function opens a file in write (append) mode for output or error writing.
+	// A nil output means that the specified path is either empty, or
+	// that something prevented the file from being opened in write mode.
```
Comment (dhaus67) [inline roxctl/common/io/io.go:43]: This most likely should be in the function description?

**Review by dhaus67 — APPROVED** [review]
LGTM modulo comments

File: roxctl/common/io/io.go (hunk 3 above)
Comment (rhybrillou) [reply roxctl/common/io/io.go:63]: Good catch.

**Pull request merged**